src/tool/cpp/strings/base_activation.h

```diff
@@ -21,6 +19,11 @@ WINRT_EXPORT namespace winrt
     }
 }
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
>Wdeprecated- [](start = 35, length = 12)

what was deprecated?

Reply:
Clang has deprecated _ReadWriteBarrier but VC has not. Previously we disabled the warning for the entire header. Here I'm just limiting it to these two functions. Separately I'll see about replacing it with a standard alternative.

Reply:
Check out the PAL's atomic_reference_count. I think I've demonstrated that we can replace all that with features from the <atomic> header.

Reply:
In general yes, but activation needs 128 bit atomics, which I don't think are available in a standard form. Some of this stuff can be made more portable. For now I'm just shooting for parity. Once I have parity with C++/WinRT I'll switch gears and look at portability again.